serdaSimus/assignments/slidingwindow/lc53/lc53.js

```diff
@@ -1,7 +1,59 @@
+// Problem Statement #
+// Given an array of positive numbers and a positive number ‘k,’ find the maximum sum of any contiguous subarray of size ‘k’.
+
+/*
+  [] ... [1] ... [1,2,3,4,5]
+  I need accumulator: sum
+  I need var: largestValue
+  I need a counter: 
+
+  for loop: 
+    add each item to sum
+    check if counter >= k:
+      compare sum and largest: if sum is bigger:
+        i would save sum to larges, else I would continue
+        sum = sum - arr[counter - k]
+    counter++
+
+  return largestValue
+      
+*/
+// [1,2,3,4,6] k=3
+
 function max_subarray_size_k(k, arr) {
-  // TODO: Write your code here
+  // basic edge cases, k is null, arr is null,
+  // k is negative, k>arr.length
+  if (!arr.length || k > arr.length || k < 0 || k == null) return null;
+
+  let sum = 0; // accumulator
+  let largest = 0; // stores hightest num
+  let counter = 1; // current position of num in a for...of loop.
+
+  // performs like a do...while
+  for (let num of arr) {
+    // limits outOfBound and offByOne errors.
+    sum += num; // accumulating the sum of the numbers visited
+
+    // checking if counter has reached a window size k
+    if (counter >= k) {
+      // comparing sum to largest, and storing larger of the two.
+      // this works because our value will never be -negative.
+      // largest = sum > largest ? sum : largest;
+
+      sum -= arr[counter - k]; // removing the first item from the last window of k size.
+    }
+    counter++; // increasing counter, effectively moving the window forward.
+  }
+  return largest;
 }
```

Comment:
MAKE IT NICE, TYPE IT TWICE! Repitition is key. Great work on adding the comments inbetween code for yourself but this was probably best as part of the high level overview inside multi-line comment above the funct.

Good work so far!